src/Physics/Multinucleon/XSection/MECScaleVsW.cxx

```diff
@@ -0,0 +1,166 @@
+//_________________________________________________________________________
+/*
+ Copyright (c) 2003-2020, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+ Original code contributed by J.Tena and M.Roda
+ Substantial code refactorizations by the core GENIE group.
+*/
+//_________________________________________________________________________
+
+#include "Framework/Algorithm/AlgConfigPool.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Physics/Multinucleon/XSection/MECScaleVsW.h"
+#include "Framework/Utils/StringUtils.h" 
+#include "Framework/ParticleData/PDGCodes.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+
+using namespace genie;
+
+//_________________________________________________________________________
+MECScaleVsW::MECScaleVsW() : 
+  XSecScaleMap()
+{
+  
+}
+//_________________________________________________________________________
+MECScaleVsW::MECScaleVsW(string config) : 
+  XSecScaleMap(config) 
+{
+  
+}
+//_________________________________________________________________________
+MECScaleVsW::~MECScaleVsW()
+{
+
+}
+//_________________________________________________________________________
+void MECScaleVsW::Configure(const Registry & config)
+{
+    Algorithm::Configure(config);
+    this->LoadConfig();
+}
+//____________________________________________________________________________
+void MECScaleVsW::Configure(string config)
+{
+    Algorithm::Configure(config);
+    this->LoadConfig();
+}
+//_________________________________________________________________________
+double MECScaleVsW::GetScaling( const Interaction & interaction ) const
+{
+  double Q0 = interaction.Kine().GetKV(kKVQ0) ; 
+  double Q3 = interaction.Kine().GetKV(kKVQ3) ; 
+
+  return GetScaling( Q0, Q3 ) ; 
+}
+//_________________________________________________________________________
+double MECScaleVsW::GetScaling( const double Q0, const double Q3 ) const
+{
+  // Get the vectors that include the kinematic limits of W for a given event:
+  std::vector<double> W_limits, weights ; 
```

Comment:
The map, But I also want is to be defined in a way the explicitly shows that the type is the same as the object defined in the header